Web Inspector: Replace the audit tests to rem instead of px

francescorn opened the pull request:
#### a4db951c083c0e1adc555a91b7402cc92faf315f
<pre>
Web Inspector: Replace the audit tests to rem instead of px
<a href="https://bugs.webkit.org/show_bug.cgi?id=267311">https://bugs.webkit.org/show_bug.cgi?id=267311</a>

Reviewed by NOBODY (OOPS!).

Replace px with rem for `font-size` in the audit tests.

* Source/WebInspectorUI/UserInterface/Views/AuditNavigationSidebarPanel.css:
(.content-view.tab.audit .content-view &gt; .audit-version):
* Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:
(.content-view.audit-test-case &gt; header &gt; .metadata &gt; .duration):
(.content-view.audit-test-case &gt; section table &gt; tr &gt; td:first-child):
* Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
(.content-view-container &gt; .content-view.audit-test &gt; header h1):
(.content-view-container &gt; .content-view.audit-test &gt; header p):
* Source/WebInspectorUI/UserInterface/Views/AuditTestGroupContentView.css:
(.content-view.audit-test-group &gt; header &gt; .information &gt; p):
(.content-view.audit-test-group &gt; header &gt; .percentage-pass):
(.content-view.audit-test-group &gt; header &gt; .percentage-pass &gt; span):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a4db951c083c0e1adc555a91b7402cc92faf315f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37568 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16452 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39830 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40097 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33462 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19077 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13616 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31850 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38134 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13860 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32932 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12078 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12042 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33618 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41362 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33951 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34037 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37941 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12621 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10169 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36108 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14062 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13023 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13375 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso reviewed (changes requested):
we explicitly prefer using `px` wherever possible as it's significantly easier to reason about the result than decimal values based on external constraints like `0.1875em`

i can understand making `font-size` use `em`, but things like `border` should rarely (if ever) be based on the `font-size` as they're purely decorative (as opposed to something the user needs to be able to read)

even then, i dont think there is a way to adjust the default `font-size` of Web Inspector in the first place, so using `em` doesn't get us any benefit if the user changes their default font size in Safari's preferences

cookiecrook commented on Source/WebInspectorUI/UserInterface/Views/AuditTestGroupContentView.css:
```diff
@@ -28,8 +28,8 @@
 .content-view.audit-test-group > section > .audit-test-group > header {
-    margin-top: -1px;
-    border-top: 1px solid var(--border-color);
+    margin-top: 0.0625em;
```
Why go from a negative 1px margin to a positive expandable one? I can't foresee a problem, but IMO this won't make much difference for low vision users.

francescorn replied:
It can stay negative, was a typo.

cookiecrook replied:
then the same question applies from https://github.com/WebKit/WebKit/pull/22633#discussion_r1449360307

cookiecrook commented on Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:
```diff
@@ -36,14 +36,14 @@
     z-index: var(--z-index-header);
     background-color: var(--audit-test-header-background-color);
-    -webkit-backdrop-filter: blur(20px);
+    -webkit-backdrop-filter: blur(1.25em);
```
We do a background color average (effectively an infinite gaussian) on iOS native with reduced transparency, but I don't recall having thought about making the blur radius relative to the font size. Cool idea!

cookiecrook commented on Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:
```diff
@@ -36,14 +36,14 @@
 .content-view-container > .content-view.audit-test-case.manager-editing > header h1 > img {
     width: 0.75em;
-    min-width: 12px;
+    min-width: 0.75em;
```
`rem` may be better than `em` on most of these, since it's not susceptible to the cascade. In this case, the img in this heading will be a multiple of the base size, times the heading text size, times the image size. The heading size may also be compounded by ancestor `em` sizes making it unpredictable if certain unrelated style changes snuck in. I think changing these to `rem` across the board avoids that unpredictability.

cookiecrook commented on Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
```diff
@@ -134,20 +134,20 @@
-    margin-inline-end: 2px;
-    vertical-align: -1px;
+    margin-inline-end: .125em;
+    vertical-align: -0.0625em;
```
Won't the negative relative size potentially cause the contents to moving farther outside the element's own box as the font size increases?

cookiecrook replied:
It may not if there's a negative corresponding margin that is also sized relatively, but I wasn't sure if that had been incorporated elsewhere.

francescorn replied:
After testing both of the `vertical-align`s as one positive and one negative or both, the best solution was to keep this `vertical-align` positive and the other `vertical-align` negative to keep the elements in alignment.

francescorn committed:
Web Inspector: Replace the audit tests to rem instead of px
https://bugs.webkit.org/show_bug.cgi?id=267311

Reviewed by NOBODY (OOPS!).

Replace px with rem for `font-size` in the audit tests.

* Source/WebInspectorUI/UserInterface/Views/AuditNavigationSidebarPanel.css:
(.content-view.tab.audit .content-view > .audit-version):
* Source/WebInspectorUI/UserInterface/Views/AuditTestCaseContentView.css:
(.content-view.audit-test-case > header > .metadata > .duration):
(.content-view.audit-test-case > section table > tr > td:first-child):
* Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
(.content-view-container > .content-view.audit-test > header h1):
(.content-view-container > .content-view.audit-test > header p):
* Source/WebInspectorUI/UserInterface/Views/AuditTestGroupContentView.css:
(.content-view.audit-test-group > header > .information > p):
(.content-view.audit-test-group > header > .percentage-pass):
(.content-view.audit-test-group > header > .percentage-pass > span):

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/AuditTestContentView.css:
```diff
@@ -34,11 +34,11 @@
 
 .content-view-container > .content-view.audit-test > header h1 {
-    font-size: 2em;
+    font-size: 2rem;
```
do we know for sure that `2em` will always be equal to `2rem` in every circumstance here?

ditto with all other `em` to `rem` changes below

francescorn replied:
Do we reference editor's drafts or working drafts?

According to the [working draft](https://www.w3.org/TR/css-values-4/#rem): rem is equal to the computed value of the [em](https://www.w3.org/TR/css-values-4/#em) unit on the root element.

It is the same in the [editor's draft](https://drafts.csswg.org/css-values-4/#font-relative-lengths): rem is equal to the computed value of the em unit on the root element.

francescorn replied:
`Command +/-` is implemented by the web inspector. That is why I changed my mind. It zooms in on the px.

dcrousso replied:
my question is more about this particular situation than the behavior of `em` vs `rem`

for example, consider

```html
<div id="a" style="font-size: 20px">
  <div id="b" style="font-size: 2em"></div>
</div>
```

currently the computed `font-size` of `#b` is `40px`, but if we changed the `font-size` of `#b` to be `2rem` instead it'd be `32px` (since the default root `font-size` is `16px`)

so my question was more about whether we're sure that the `font-size` of every parent of every element matching `content-view-container > .content-view.audit-test > header h1` is the same as the root `font-size`, or if there's a scenario where some parent sets a `font-size` that's different from the root `font-size` (which would cause `2em` to compute to a different value than `2rem`)?

dcrousso reviewed (changes requested):
if there's no way to change the default root `font-size`, then there's really no difference between `px` and `rem`, so why is this change necessary?

> Drafting until `Cmd Opt +/-` is in the web inspector.

is this no longer the plan?
